File: RadVolontera/RadVolontera.Models/VolunteeringAnnouncement/VolunteeringAnnouncement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RadVolontera.Models.VolunteeringAnnouncement
{
    public class VolunteeringAnnouncement
    {
        public long Id { get; set; }
        public string Place { get; set; }
        public string? Reason { get; set; }
        public long AnnouncementStatusId { get; set; }
        public Models.Status.Status AnnouncementStatus { get; set; }
        public long CityId { get; set; }
        public Models.City.City City { get; set; }
        public string MentorId { get; set; }
        public Models.Account.UserResponse Mentor { get; set; }
        public string TimeFrom { get; set; }
        public string TimeTo { get; set; }
        public DateTime Date { get; set; }
        public string Notes { get; set; }
        public bool HasReport { get; set; } = false;
    }
}

File: RadVolontera/UI/radvolontera_admin/lib/models/volunteering_announcement/volunteering_announcement.dart



import 'package:json_annotation/json_annotation.dart';
import 'package:radvolontera_admin/models/account/account.dart';
import 'package:radvolontera_admin/models/city/city.dart';
import 'package:radvolontera_admin/models/status/status.dart';

part 'volunteering_announcement.g.dart';
@JsonSerializable()
class VolunteeringAnnouncementModel {
  int? id;
  String? place;
 int? announcementStatusId;
  StatusModel? announcementStatus;
  int? cityId;
  CityModel? city;
  String? mentorId;
  AccountModel? mentor;
  String timeFrom;
  String timeTo;
 String? notes;
 DateTime ? date;
 String ? reason;
VolunteeringAnnouncementModel(this.id, this.place, this.announcementStatusId, 
this.announcementStatus,this.cityId,this.city,this.mentorId,this.mentor,this.timeFrom,this.timeTo,this.reason);
      /// A necessary factory constructor for creating a new User instance
  /// from a map. Pass the map to the generated `_$UserFromJson()` constructor.
  /// The constructor is named after the source class, in this case, User.
   factory VolunteeringAnnouncementModel.fromJson(Map<String, dynamic> json) => _$VolunteeringAnnouncementModelFromJson(json);

  // /// `toJson` is the convention for a class to declare support for serialization
  // /// to JSON. The implementation simply calls the private, generated
  // /// helper method `_$UserToJson`.
   Map<String, dynamic> toJson() => _$VolunteeringAnnouncementModelToJson(this);
}